Endava.TechCourse.BankApp.Application/Commands/CreateWallet/CreateWalletCommandHandler.cs

```diff
@@ -1,42 +1,60 @@
 ﻿using Endava.TechCourse.BankApp.Domain.Models;
 using Endava.TechCourse.BankApp.Infrastructure.Persistence;
+using Endava.TechCourse.BankApp.Infrastructure.Services;
 using MediatR;
+using Microsoft.EntityFrameworkCore;
 
 namespace Endava.TechCourse.BankApp.Application.Commands.CreateWallet
 {
 	public class CreateWalletCommandHandler : IRequestHandler<CreateWalletCommand, CommandStatus>
 	{
 		private readonly ApplicationDbContext _context;
+		private readonly WalletCodeGenerator _walletCodeGenerator;
 
-		public CreateWalletCommandHandler(ApplicationDbContext context)
+		public CreateWalletCommandHandler(ApplicationDbContext context, WalletCodeGenerator walletCodeGenerator)
 		{
+			ArgumentNullException.ThrowIfNull(context);
+			ArgumentNullException.ThrowIfNull(walletCodeGenerator);
+
 			_context = context;
+			_walletCodeGenerator = walletCodeGenerator;
 		}
 
 		public async Task<CommandStatus> Handle(CreateWalletCommand request, CancellationToken cancellationToken)
 		{
-			if (!_context.Users.Any(u => u.Id.ToString() == request.OwnerId))
-			{
-				return CommandStatus.Failed("User doesn't exists");
-			}
+			if (!_context.Users.Any(u => u.Id.ToString() == request.OwnerId)) return CommandStatus.Failed("User doesn't exists");
```

Comment:
Is better to assigne a value before If() statement and to use the Async method if it's possible :
```
var status = _context.Users.AnyAsync(u => u.Id.ToString() == request.OwnerId, cancellationToken);
if (status){}
```